Implement dicom_to_nifti: wrapper for dcm2bids to convert DICOM data

gaspardcereza opened the pull request:
## Context
The conversion from dicom to nifti is one of the first step of our workflow. We thus need to have a functional code to allow us to organize our data in the format we want. The former Matlab functions we use must be adapted to the python standards.

This PR introduces the following changes:
- shimmingtoolbox/dicom_to_nifti.py → API
- config/dcm2bids.json --> config file that lists the sequence names used for organizing data under BIDS directory. This file should be edited. https://github.com/shimming-toolbox/shimming-toolbox-py/issues/32

BIDS convention for file naming: https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#task-including-resting-state-imaging-data

dcm2bids documentation about conversion and config file:
- https://github.com/cbedetti/Dcm2Bids/issues/26
- config file for phasediff data: https://github.com/cbedetti/Dcm2Bids/issues/21#issuecomment-369943143

TODO in subsequent PRs:
- cover realistic scenario of experimental workflow. #31
- A function `read_nii()` initially developed here is no more used, but code could be recycled: https://github.com/shimming-toolbox/shimming-toolbox-py/issues/33

Fixes #12.

jcohenadad commented on Shim_revC/ShimOpt_revC.m:
```diff
@@ -0,0 +1,257 @@
+Params.dataLoadDirectories = cell( Params.nEchoes, 2, Params.nCurrents, Params.nChannels ) ;
+
+tmp = { ...
+    '/Users/ryan/Projects/Shimming/Acdc/Calibration/data/acdc_60p/33-fm2d_CH1_+0.5A/' ;
```
hardcoded path-- will it work?

rtopfer replied:
Few options:
1. Hardcode, but use Duke paths (this seems most reasonable to me for now, until we have "the best" shim reference maps such that we don't care to reacquire them)
2. Include all the DICOMs in the Google drive (each data set could be ~1Gb, and we may want several data sets...)
3. Host them somewhere else...
That being said, why is there a ShimOpt "_revC"?

jcohenadad commented:
@rtopfer can we consider merging this branch?

alexfoias commented:
@jcohenadad I think we can merge this branch, we've tested it in the past. If we want to optimize more we can do it later.

gaspardcereza committed:
started to convert nifti_read and dicom_to_nifti into python

gaspardcereza committed:
deleted rescaling steps in read_nii because nibabel does it automatically

gaspardcereza committed:
added dataset downloading from OSF

jcohenadad committed:
Replaced OSF with github link

jcohenadad committed:
Sorting out data download and path naming

jcohenadad committed:
Added docstrings, minor comments

gaspardcereza committed:
Changed some variable names, completed dicom_to_nifti

gaspardcereza committed:
updated variable names and error handling

gaspardcereza committed:
Prevent git checking in sample dataset

gaspardcereza committed:
Uptaded functions according to reviews

gaspardcereza committed:
modified config.json files generation

gaspardcereza committed:
Added possibility to not read the images when calling read_nii

jcohenadad committed:
Added test (unfinished)

jcohenadad committed:
Added download testing data in Travis

jcohenadad committed:
Removed one "script" instance

jcohenadad committed:
added click as dep

jcohenadad committed:
Added testing_data to .gitignore

jcohenadad committed:
Forgot comma

jcohenadad committed:
Added __dir_testing__ to be called by unit tests

jcohenadad committed:
Implemented unit test for dicom_to_nifti()

jcohenadad committed:
Removed examples/script_nifti.py

Now replaced by tests/test_dicom_to_nifti

jcohenadad committed:
Changed variable name

jcohenadad committed:
Added dcm2bids as a dependency

gaspardcereza committed:
deleted rescaling function (now included in read_nii)

gaspardcereza committed:
removed .gitignore in /examples (testing data no more uploaded there)

gaspardcereza committed:
updated read_nii

gaspardcereza committed:
Modified dicom_to_bids to remove .json in nifti path + updated unit test

gaspardcereza committed:
corrected errors in download functions

gaspardcereza committed:
added tests to check that json files are created

gaspardcereza committed:
added test for version of dcm2bids

jcohenadad committed:
Removed test of dcm2bids version; now specified in setup

jcohenadad committed:
Refactored dicom_to_nifti

Direct use of dcm2bids with a config .json file

jcohenadad committed:
Added dcm2bids.json config file

jcohenadad committed:
Updated config file to output BIDS-compatible fieldmap

jcohenadad committed:
Working unit test

jcohenadad committed:
Cleanup

jcohenadad committed:
Removed function not used here

Also see: https://github.com/shimming-toolbox/shimming-toolbox-py/issues/33

gaspardcereza committed:
deleted files

kousu committed:
Upgrade dcm2niix to latest.

dcm2bids has only ever been tested with the latest version,
and dcm2niix is still shaking out its API bugs, so the only
reliable thing to do is to demand everyone stay up to date.

This is okay almost everywhere except on Debian which is always
slow to upgrade stuff.

kousu committed:
Put my versioning snippet back together.

kousu reviewed (approved):
This turned into a pretty solid little PR. It says it's a wrapper for dcm2bids, and indeed it contains:

- shimmingtoolbox/dicom_to_nifti.py - the implementation
- config/dcm2bids.json - supporting file for the implementation
- tests/test_dicom_to_nifti.py - a test for the implementation

My one hold-out concern is the bigger issue that I don't really understand this "config" file (it's really a program in miniature) as I mentioned https://github.com/shimming-toolbox/shimming-toolbox-py/pull/18/files#r462836479. But I think it's a good start and we can figure that out as we go along in further PRs.
